src/main/java/ticTacToe/ui/Ui.java

```diff
@@ -94,33 +110,33 @@ private boolean isInvalidOptionNumber(String opponentOptionNumber) {
         return !opponentOptionNumber.equals("1") && !opponentOptionNumber.equals("2");
     }
 
-    private String validMarkType(String mark) {
-        while (isInvalidMark(mark)) {
-            output.println("Invalid option: X => Cross or O (letter) => Nought");
+    private Mark validMarkType(String mark) {
+        while (!Mark.isValidMark(mark)) {
+            clearScreen();
+            output.println("Invalid option: " + GridPreparer.blue_X + " => Cross or " + GridPreparer.red_O + " (letter) => Nought");
             mark = input.nextLine().toUpperCase();
         }
-        return mark;
+        return Mark.createMark(mark);
     }
 
-    private boolean isInvalidMark(String mark) {
-        return !CROSS.mark.equals(mark) && !NOUGHT.mark.equals(mark);
-    }
-
-    private String validatePosition(Grid grid, String mark, Lines lines, String usersInput) {
+    private String validatePosition(Grid grid, Mark mark, Lines lines, String usersInput) {
         while (isOutsideValidRange(usersInput, grid)) {
+            clearScreen();
             output.println("Invalid position.");
             usersInput = promptForNewInput(grid, mark, lines);
+            printGrid(lines, grid);
         }
         return usersInput;
     }
 
     private boolean isOutsideValidRange(String gridPosition, Grid grid) {
         Integer positionNumber = Integer.parseInt(gridPosition);
-        return positionNumber < 0 || positionNumber > grid.getCells().size();
+        return positionNumber <= 0 || positionNumber > grid.getCells().size();
```

Comment:
Here you are trying to access the data from the Grid (Board?)
Wouldn't it be better if you moved this method to this class?